boa/src/builtins/value/mod.rs

```diff
@@ -616,19 +608,17 @@ impl ValueData {
 
     /// Check whether an object has an internal state set.
     pub fn has_internal_state(&self) -> bool {
-        if let Self::Object(ref obj) = *self {
-            obj.borrow().state.is_some()
-        } else {
-            false
+        match self.as_object() {
+            Some(object) => object.state().is_some(),
+            None => false,
         }
```

Comment:
I'm not sure if a `.map().unwrap_or()` would be easier to understand, what do you think?

Reply:
we could use `std::matches!()` macro

Reply:
> we could use `matches!()` macro

Yes!